zilliqa/src/exec.rs

```diff
@@ -553,17 +588,35 @@ impl State {
         let hash = txn.hash;
         let from_addr = txn.signer;
         info!(?hash, ?txn, "executing txn");
+        let mut is_scilla = false;
+
+        match txn.tx {
+            SignedTransaction::Zilliqa { .. } => {
+                info!("Zilliqa transaction detected, using zilliqa interpreter...");
+                is_scilla = true;
+            }
+            _ => {}
+        }
 
         let txn = txn.tx.into_transaction();
 
         // fail early on nonce mismatch
         let mut acct = self.get_account(from_addr).unwrap();
-        if acct.nonce != txn.nonce() {
+
+        let desired_account_nonce = if is_scilla {
```

Comment:
The js framework seems to be sending a tx starting from nonce 0 for some reason, need to investigate...